test/Microsoft.IdentityModel.Tests/ReferenceXml.cs

```diff
@@ -638,6 +638,239 @@ public static SamlActionTestSet SamlActionValid
         }
         #endregion
 
+        #region SamlAdvice
+        public static SamlAdviceTestSet AdviceNoAssertionIDRefAndAssertion
+        {
+            get
+            {
+                return new SamlAdviceTestSet
+                {
+                    Advice = new SamlAdvice(),
+                    Xml = XmlGenerator.SamlAdviceXml(null, null)
+                };
+            }
+        }
+
+        public static SamlAdviceTestSet AdviceWithAssertionIDRef
+        {
+            get
+            {
+                return new SamlAdviceTestSet
+                {
+                    Advice = new SamlAdvice(new string[] { Default.SamlAssertionID }),
+                    Xml = XmlGenerator.SamlAdviceXml(XmlGenerator.SamlAssertionIDRefXml(Default.SamlAssertionID), null)
+                };
+            }
+        }
+
+        // TODO : Add this test case after complete SamlAssertion test cases
+        public static SamlAdviceTestSet SamlAdviceWithAssertions
+        {
+            get
+            {
+                return new SamlAdviceTestSet
+                {
+                    Advice = new SamlAdvice(new List<SamlAssertion> { SamlAssertionNoSignature.Assertion }),
+                    Xml = XmlGenerator.SamlAdviceXml(null, SamlAssertionNoSignature.Xml)
+                };
+            }
+        }
+
+        public static SamlAdviceTestSet SamlAdviceWithWrongElement
+        {
+            get
+            {
+                return new SamlAdviceTestSet
+                {
+                    Advice = new SamlAdvice(new string[] { Default.SamlAssertionID }),
+                    Xml = XmlGenerator.SamlAdviceXml(SamlActionValid.Xml, null)
+                };
+            }
+        }
+
+        public static SamlAdviceTestSet SamlAdviceWithAssertionIDRefAndAssertions
+        {
+            get
+            {
+                return new SamlAdviceTestSet
+                {
+                    Advice = new SamlAdvice(new string[] { Default.SamlAssertionID }, new List<SamlAssertion> { SamlAssertionNoSignature.Assertion }),
+                    Xml = XmlGenerator.SamlAdviceXml(XmlGenerator.SamlAssertionIDRefXml(Default.SamlAssertionID), SamlAssertionNoSignature.Xml)
+                };
+            }
+        }
+        #endregion
+
+        #region SamlAssertion
+        public static SamlAssertionTestSet SamlAssertionMissMajorVersion
+        {
+            get
+            {
+                return new SamlAssertionTestSet
+                {
+                    Assertion = new SamlAssertion(Default.SamlAssertionID, Default.Issuer, DateTime.Parse(Default.IssueInstant), SamlConditionsSingleCondition.Conditions, AdviceWithAssertionIDRef.Advice, new List<SamlStatement> { SamlAttributeStatementSingleAttribute.AttributeStatement }),
```

Comment:
In general, negative tests don't need a fully populated object to compare against.
It would be good to set them to null so that the test will certainly fail.